### DISCUSSION TITLE
Regarding magneto-optical drive models, capacity limits and features

### DISCUSSION THREAD

**Discussion by ppgrainbow** [86Box/86Box#1230] [General]
I would like to post a question regarding the emulation of magneto-optical (MO) drives.

I have found some information regarding the model name, maximum native capacity and the interface that it uses. The following MO drives that 86Box emulates only uses a SCSI interface and on seven of the emulated models listed below, there is information about the model name below:

Fujitsu M2512A 230 MB SCSI 3.5" -> https://archive.org/stream/bitsavers_fujitsuFuj_2862594/Fujitsu_M2512A_LR_djvu.txt
Fujitsu M2513-MCC3064SS 640 MB SCSI 3.5" -> https://www.cnet.com/products/fujitsu-mcc3064ss-mo-disk-drive-scsi-ca01760b105/
Fujitsu MCE3130SS 1.3 GB SCSI 3.5" -> https://www.cnet.com/products/fujitsu-mce-3130ss-mo-disk-drive-scsi-internal/
Fujitsu MCF3064SS 640 MB SCSI 3.5" -> https://www.cnet.com/products/fujitsu-magneto-optical-mcf3064ss-internal/
Fujitsu MCJ3230UB-S 2.3 GB 3.5"

Hewlett Packard S6300.65 (unknown)
Hewlett Packard C1716C 650 MB SCSI 5.25"

IBM 0632-AAA - 650 MB SCSI
IBM 0632-CHC - 1.3 GB SCSI 5.25"
IBM 0632-CHX - 1.3 GB SCSI
IBM MD3125-A - 128 MB SCSI 3.5"
IBM MD3125-B - 128 MB SCSI 3.5"
IBM MTA-3127 - 128 MB SCSI 3.5"
IBM MTA-3230 - 230 MB SCSI 3.5"

Panasonic (Matsushita) LF-3000 - 128 MB SCSI 3.5"

Mass Optical Storage Technologies (MOST) RMD-5100-S - 128 MB SCSI 3.5" -> https://www.questia.com/magazine/1G1-9620789/most-intros-rewritable-magneto-optic-disk-drive-using

Ricoh RO-5031E - 650 MB SCSI 3.5"

Sony SMO-C301 - 9.1 GB SCSI 3.5" -> https://www.cnet.com/products/sony-smo-f561-mo-disk-drive-scsi-internal/
Sony SMO-C501 - 650 MB SCSI 5.25"

Teac OD-3000 - 128 MB SCSI 3.5" -> https://books.google.com/books?id=rT0EAAAAMBAJ&pg=PA26&lpg=PA26&dq=%22Teac+OD-3000%22&source=bl&ots=uXyGvvGya6&sig=ACfU3U0U1yy3ks_0e22D3eBvDRNXHEHWzw&hl=en&sa=X&ved=2ahUKEwjzk-zb6trqAhU4CTQIHUcXABIQ6AEwBnoECAoQAQ#v=onepage&q=%22Teac%20OD-3000%22&f=false

Toshiba OD-D300 - 230 MB SCSI 3.5"

(The figures next to each model name is the maximum native capacity size, interface and height of the drive - either 3.5" or 5.25".)

For the remaining 14 MO models, I can't seem to find information regarding the specs on the drives. Can anyone find some information and the specs on the remaining 14 MO models?

Also, in the "Other Removable Devices" section under the Type drop-down menus, I think that it would good to specify the native maximum capacity of the MO model that the users specifies and exclude larger capacities that exceed the maximum native capacity of the selected MO model in the Disk size drop down box of the new MO image dialog box.

Lastly in the Type drop-down menu section of the "Other Removable Devices" in 86Box settings, there is a typo that reads "MATSHITA LF-3000 1.00". It's supposed to read "PANASONIC (MATSUSHITA) LF-3000 128 MB".

I'm sorry if my grammar isn't 100% perfect, but I've been trying to provide as much information as possible regarding the magneto-optical drives.

**Comment by claunia** [comment]
Hi @ppgrainbow,

I checked a lot of places to find the limitations of those drive models.

Most of them are SCSI but Fujitsu specially used the same mechanism in ATAPI and USB buses for many drives.

The idea of having the models is for some software that cares about the drive identifier, but do not care if the drive is SCSI, ATAPI, or USB, as those three buses will appear as SCSI to them anyways. I have used a USB drive in eComStation with the IBM driver for OS/2 2.0, for the driver it was just ASPI.

As for the media, there are two types of media basically:
1.- ISO
2.- Non-ISO

For 3.5" it is easily known, 128, 230, 540 and 640 are both ISO, 1.3 and 2.3Gb are non-ISO by Fujitsu.
For 5.25" there is a whole lot of non-ISO <2Gb, plus some ISO, and most >2Gb are ISO.

The media sizes in the ISO standards is incorrect, as it counts the reserved sectors and others non accessible by the user.

So the media sizes defined in code are the ones for which I have media and tested them.

Again on 3.5" it is easy, all drives support all media below them.
5.25" I have filled as I have found documentation, or a working drive. It is not a "max capacity and supports all below", it is a table of compatibility.
You can see many drives support a 1.2Gb 1k MO, but don't support the 1Gb 512b MO.
Worse even some drives are natively 4k (or 2k, or 1k) and support media that is not, reporting it is 4k (this is not yet implemented).

The idea was to have the following modes:
1.- Specific model with strict media compatibility: That is support only the media that the drive really support.
2.- Specific model with non-strict media compatibility: That is support any media, even if the drive does not really support it, as most software that has any kind of restriction, only looks at the drive identification, not the media characteristics itself.
3.- Non specific model: A generic identification that supports any media.

**Comment by ppgrainbow** [comment]
Thank you very much for your suggestions. I really appreciate it.

With your three suggestions, we're hopeful that the magneto-optical emulation in 86Box will improve even more. :D

**Comment by claunia** [comment]
Whenever I get more media and drives I will update the tables :D

**Comment by ppgrainbow** [comment]
Thank you for the heads up. It would be good to include ATAPI-based magneto-optical drives as well. ^^

**Comment by rainlander** [comment]
I have a Fujitsu M2511A (128 MB), a Fujitsu MCD3130SS (1.3 GB) and some 128 MB, 230 MB, 540 MB, 640 MB and 1.3 GB MO disks ready to be thrown out. If you want some information about them (copy of the manuals, number of sectors on the disks etc), let me know.

**Reply by ppgrainbow** [reply]
> 
> 
> I have a Fujitsu M2511A (128 MB), a Fujitsu MCD3130SS (1.3 GB) and some 128 MB, 230 MB, 540 MB, 640 MB and 1.3 GB MO disks ready to be thrown out. If you want some information about them (copy of the manuals, number of sectors on the disks etc), let me know.

Sure! I would like some more information on these magneto-optical drive models. :)

**Reply by rainlander** [reply]
Ok, let me know what you need and how to contact you, and I'll take it from there.

**Reply by Bruce727Home** [reply]
Hi Rainlander.  If you haven't yet discarded your MO drives and media, but wish to unload it, I'd like to acquire all of it to support legacy Mac systems I'm restoring, and would be happy to pay the shipping or provide my FedEx account number, and some extra for your time to package it.  Please let me know if interested!  (Here or AirplaneHome.com has my direct contact information.)  Many thanks!

**Comment by Bruce727Home** [comment]
I own a Ricoh RO-5031E with lots of 650 MB media, plus a Pinnacle Micro OHD-2600 Vertex 2.6 GB, but have only 1.3 MB media for it, which thus far seems incompatible.  I have no specs for either but sent a request to Ricoh for RO-5031E documentation and the last firmware version plus an installation utility (I've no idea whether that'll bear useful results).  The device served perfectly for years, then was shelved, but now I'm trying to put it back into service, thus far unsuccessfully for reasons unknown, though I fear firmware data fade out.

If I can help please let me know (AirplaneHome.com has my contact information).

Rainlander, in the meantime if you intend to dumpster pitch your magneto optical items but would consider shipping them to me instead I'd be grateful, and pay at least the economy shipping cost of course.  Cheers everyone!
